just_audio/darwin/Classes/AudioPlayer.m

```diff
@@ -737,7 +737,7 @@ - (void)onComplete:(NSNotification *)notification {
         _index = [_order[([_orderInv[_index] intValue] + 1) % _order.count] intValue];
         [self broadcastPlaybackEvent];
         _justAdvanced = YES;
-    } else if ([_orderInv[_index] intValue] + 1 < [_order count]) {
+    } else if (_index < [_orderInv count] && [_orderInv[_index] intValue] + 1 < [_order count]) {
```

Comment:
Tentei mais algumas coisas pra reproduzir esse crash mas em vão...

Estou pensando aqui: se o `_index` for inválido, em vez de crashar vai car no `else` que define o estado do player como `complete`. Estou achando mais safe não fazer absolutamente nada quando o `_index` for inválido, na esperança do estado, eventualmente, ser atualizado corretamente nas próximas atualizações do AvPlayer.